modelscope_agent/llm/openai.py

```diff
@@ -38,21 +65,30 @@ def _chat_stream(self,
         logger.info(
             f'call openai api, model: {self.model}, messages: {str(messages)}, '
             f'stop: {str(stop)}, stream: True, args: {str(kwargs)}')
-        stream_options = {'include_usage': True}
-        response = self.client.chat.completions.create(
-            model=self.model,
-            messages=messages,
-            stop=stop,
-            stream=True,
-            stream_options=stream_options,
-            **kwargs)
+        if not self.is_azure:
+            stream_options = {'include_usage': True}
+            response = self.client.chat.completions.create(
+                model=self.model,
+                messages=messages,
+                stop=stop,
+                stream=True,
+                stream_options=stream_options,
```

Comment:
why azure api and openai api are processed differently, they should be identical in terms of api interfaces (for example, both support stream_options)

Reply:
- This is because the Azure API is not always [fully compatible](https://github.com/openai/openai-python?tab=readme-ov-file#microsoft-azure-openai) with the OpenAI API. It depends on the `api_version` of the Azure instance being used, which in my case is `2024-06-01`. 
- When I make a request, it results in an error: `openai.BadRequestError: Error code: 400 - {'error': {'message': "Unknown parameter: 'stream_options'.", 'type': 'invalid_request_error', 'param': 'stream_options', 'code': 'unknown_parameter'}}.`
-  I reviewed the official [request body](https://learn.microsoft.com/en-us/azure/ai-services/openai/reference-preview#request-body-2) and did not find the 'stream_options' parameter. 
- Therefore, it is recommended to handle the request parameters for Azure OpenAI and OpenAI separately for better compatibility.

Reply:
looks like it is ineed the case, https://github.com/openai/openai-python/issues/1469

that is very unfortunate.

still, i would recommend putting stream_options into kwargs (only when using native openai api), which would make the code much cleaner: there is not need to duplicate the entire completions.create call.

Reply:
done